### DISCUSSION TITLE
Cannot find parent element in DOM tree containing attribute: [wire:id]

### DISCUSSION THREAD

**Discussion by vrajroham** [livewire/livewire#4352] [1. Help]
Whenever a new event is fired, this error is thrown on the listener component. If I don't provide any listeners on the component, it does not show any error.

Here are my components,

```php
<! -- order-taker view-->
<div>
    @foreach($categories as $category)
        <livewire:category wire:key="{{ $loop->index }}" :category="$category"/>
    @endforeach
</div>
```

```html
<!-- category view -->
<div>
    <div
        class="..."
        wire:click="getItemsForCategory('{{$category}}')">
        <p class="...">
            {{$category}}
        </p>
    </div>
</div>
```

```php
// Category Component
class Category extends Component
{
    public $category;

    public function render()
    {
        return view('livewire.category');
    }

    public function getItemsForCategory($category)
    {
        $this->emit('categoryChanged', $category);
    }
}
```

```php
<?php

namespace App\Http\Livewire;

use Illuminate\Support\Facades\Log;
use Livewire\Component;

class OrderTaker extends Component
{
    public $categories;
    public $selectedCategory;

    public function getListeners()
    {
        return [
            'categoryChanged' => 'setSelectedCategory',
        ];
    }

    public function mount()
    {
        $this->categories = [
            'Appetizers',
            'Entrees',
            'Desserts',
        ];
    }

    public function setSelectedCategory(string $category)
    {
        $this->selectedCategory = $category;
    }

    public function render()
    {
        return view('livewire.order-taker');
    }
}
```

When I remove the listener, the error goes away.

**Comment by denizozturk** [comment]
Can you try changing the `public` value to `protected` for the `getListeners` method?

Docs for this function: https://laravel-livewire.com/docs/2.x/events#event-listeners

**Reply by vrajroham** [reply]
@denizozturk Tried changing it from `public` to `protected` but still same.

**Comment by joshhanley** [comment]
@vrajroham can you share a screenshot of the stack trace? Thanks!

**Reply by vrajroham** [reply]
@joshhanley Here is the stack trace.

![image](https://user-images.githubusercontent.com/12662173/145939312-f677b935-3cd9-4d5f-abfe-23abc2b6eb15.png)

**Comment by jonneroelofs** [comment]
Are you sure the key you use is unique on the entire page?
Maybe change it to wire:key="category_{{ $loop->index }}" to make sure .

**Comment by mairon-silverbox** [comment]
I have this problem 
Uncaught (in promise) 
Livewire Error:

Cannot find parent element in DOM tree containing attribute: [wire:id].

Usually this is caused by Livewire's DOM-differ not being able to properly track changes.
